app/src/main/java/com/android/calculator2/widget/CalculatorWidget.java

```diff
@@ -60,6 +61,9 @@ public void onReceive(Context context, Intent intent) {
         if(value.equals(context.getResources().getString(R.string.error_syntax))) value = "";
         mClearText = intent.getBooleanExtra(SHOW_CLEAR, false);
 
+        String action = intent.getAction().substring(intent.getAction().lastIndexOf(".")+1);
```

Comment:
What is an expected value to be returned via `intent.getAction`. Can you type a sample value in a comment?